src/main/java/io/reactorsolutions/rxjava/verticles/RxConsumerVerticle.java

```diff
@@ -0,0 +1,53 @@
+package io.reactorsolutions.rxjava.verticles;
+
+import io.reactivex.rxjava3.core.Completable;
+import io.reactivex.rxjava3.core.CompletableSource;
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.core.ObservableSource;
+import io.reactorsolutions.vertx_kafka.MainVertxConsumer;
+import io.reactorsolutions.vertx_kafka.config.consumer.ConsumerOptions;
+import io.vertx.core.Context;
+import io.vertx.core.Promise;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.rxjava3.core.AbstractVerticle;
+import io.vertx.rxjava3.kafka.client.consumer.KafkaConsumer;
+import io.vertx.rxjava3.kafka.client.consumer.KafkaConsumerRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+public class RxConsumerVerticle extends AbstractVerticle {
+  private static final Logger LOG = LoggerFactory.getLogger(RxConsumerVerticle.class);
+
+  private KafkaConsumer<String, JsonObject> consumer;
+
+
+  @Override
+  public void init(Vertx vertx, Context context) {
+    super.init(vertx, context);
+    consumer = KafkaConsumer.create(this.vertx, new ConsumerOptions().getConfig());
+  }
+
+  @Override
+  public Completable rxStart() {
+    return consumer.subscribe("rxJava").andThen((CompletableSource) completableObserver ->
```

Comment:
Do you need the casting if you are not actually using the param `completableObserver`?

Reply:
yes

Reply:
Update this with the last modificationss